packages/buidler-core/src/internal/cli/analytics.ts

```diff
@@ -191,44 +191,49 @@ export class Analytics {
 }
 
 async function getClientId() {
-  // TODO: Check Windows support for this approach
-  const globalBuidlerConfigFile = path.join(
-    os.homedir(),
-    ".buidler",
-    "config.json"
-  );
-
-  await fs.ensureFile(globalBuidlerConfigFile);
+  const { getDataDir } = await import("../util/global-dir");
```

Comment:
Importing buidler files with top-level imports is fine. Especially since you import the dependencies dynamically there.